src/MarkdownDeep/LinkDefinition.cs

```diff
@@ -58,9 +58,36 @@ internal void RenderLink(Markdown m, StringBuilder b, string link_text, List<str
 			{
 				HtmlTag tag = new HtmlTag("a");
 
+			    var url = this.Url;
+
+			    if (m.DocNetMode && m.ConvertLocalLinks)
+			    {
+			        // A few requirements before we can convert local links:
+                    //   1. Link contains .md
+                    //   2. Link is relative
+                    //   3. Link is included in the index
+                    var index = url.LastIndexOf(".md", StringComparison.OrdinalIgnoreCase);
+			        if (index >= 0)
+			        {
+			            Uri uri;
+			            if (Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out uri))
+			            {
+			                if (!uri.IsAbsoluteUri)
+			                {
+			                    // TODO: Check if link exists in the ToC
+			                    var existsInTableOfContents = true;
+			                    if (existsInTableOfContents)
+			                    {
+			                        url = url.Remove(index, ".md".Length).Insert(index, ".htm");
```

Comment:
```cs
url = String.Concat(url.Substring(0, index), ".htm", url.Substring(index + ".md".Length));
```
is more efficient: .Remove() will create a new string with both ends concatenated together, then the insert will destroy that string again and allocate a new one. The String.Concat() will use a recycled stringbuilder from a pool to concat string together in 1 go. :) The ".md".Length can be hardcoded, as it's a constant, but for clarity I think this is better indeed.